### PR TITLE
Add array_sort lambda Presto function 

### PR DESCRIPTION
**Pull request by mbasmanova**
Add a version of array_sort function that takes a lambda to compute sorting
keys. For example, this allows to sort an array of strings by string lengths.

```
SELECT array_sort(ARRAY ['cat', 'leopard', 'mouse'], x -> length(x)); -- ['cat', 'mouse', 'leopard']
```

This will be used to implement most common cases of a more generic array_sort
function which takes a comparator lambda:
https://prestodb.io/docs/current/functions/array.html#id2

In most cases, the comparator is a simple < or > comparison on values derived
from the array elements, e.g. a subfield of a struct.

Partial fix for #5545

### CODE REVIEW

File: velox/functions/prestosql/ArraySort.cpp [hunk 1]
```diff
@@ -273,7 +344,11 @@ std::shared_ptr<exec::VectorFunction> createTyped(
 std::shared_ptr<exec::VectorFunction> create(
     const std::vector<exec::VectorFunctionArg>& inputArgs,
     bool ascending) {
-  validateType(inputArgs);
+  if (inputArgs.size() == 2) {
+    const auto lambdaType = inputArgs[1].type;
```
Comment (xiaoxmeng) [inline velox/functions/prestosql/ArraySort.cpp:348]: lambdaType is not used?

File: velox/functions/prestosql/ArraySort.cpp [hunk 2]
```diff
@@ -248,17 +260,76 @@ class ArraySortFunction : public exec::VectorFunction {
...
+    // Compute sorting keys.
+    VectorPtr newElements;
+
+    auto elementToTopLevelRows = getElementToTopLevelRows(
+        newNumElements, rows, flatArray.get(), context.pool());
+
+    // Loop over lambda functions and apply these to elements of the base array.
+    // In most cases there will be only one function and the loop will run once.
+    auto it = args[1]->asUnchecked<FunctionVector>()->iterator(&rows);
+    while (auto entry = it.next()) {
```
Comment (xiaoxmeng) [inline velox/functions/prestosql/ArraySort.cpp:297]: Where we advance it? Thanks!

**Review by xiaoxmeng — COMMENTED** [review]
@mbasmanova LGTM. Thanks!

File: velox/functions/prestosql/ArraySort.cpp (hunk 1 above)
Comment (mbasmanova) [reply velox/functions/prestosql/ArraySort.cpp]: Nice catch. Removed.

**Comment by mbasmanova** [comment]
@xiaoxmeng Meng, thank you for review. Addressed / replied to comments. Please, take another look.

**Review by xiaoxmeng — APPROVED** [review]
@mbasmanova LGTM. Thanks for the explanation!

**Pull request closed without merging**